From 1260ebf2c767e5bd9e83b5d5148c2443348ac6c3 Mon Sep 17 00:00:00 2001
Message-Id: <1260ebf2c767e5bd9e83b5d5148c2443348ac6c3.1429601425.git.chang-joon.lee@intel.com>
In-Reply-To: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
References: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Thu, 16 Apr 2015 23:28:45 +0530
Subject: [PATCH 8/9] SQUASHME! [VPG]: drm/i915: Update wm1 to 0.

This patch updates the wm1 values to 0, as
recommended by SV.

SQUASHME: This patch needs to me squashed with
        commit e8c876f7944d03b2748c9afe355ba57d6d313e25
        Author: vandita kulkarni <vandita.kulkarni@intel.com>
        Date:   Fri Apr 10 21:10:24 2015 +0530

                REVERTME: [VPG]: drm/i915 : program dynamic dsparb and wm

Issue: GMINL-4468
Change-Id: I5ef76a021775843b9d9aa390de8db51aff1c4f4b
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |   10 ++++++++++
 drivers/gpu/drm/i915/intel_pm.c |   10 ++++++++++
 2 files changed, 20 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 6d66c6a..2104ee6 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4281,17 +4281,23 @@ enum punit_power_well {
 #define VLV_DSPFW9_DEF_WM	0x40f0004
 
 #define VLV_FW_PIPEA_PA_MASK	0xff
+#define VLV_FW_PIPEA_PA_WM1_MASK	0xff0000
 #define VLV_FW_PIPEA_SA_MASK	0xff
+#define VLV_FW_PIPEA_SA_WM1_MASK	0xff
 #define VLV_FW_PIPEA_SB_MASK	0xff0000
+#define VLV_FW_PIPEA_SB_WM1_MASK	0x00ff0000
 #define VLV_FW_PIPEA_SB_SHIFT	16
 #define VLV_DSPHOWM_PIPEA_MASK	0x111
 #define VLV_DSPHOWM_PIPEA_SA_SHIFT	4
 #define VLV_DSPHOWM_PIPEA_SB_SHIFT	8
 
 #define VLV_FW_PIPEB_PB_MASK	0xff00
+#define VLV_FW_PIPEB_PB_WM1_MASK	0xff000000
 #define VLV_FW_PIPEB_PB_SHIFT	8
 #define VLV_FW_PIPEB_SC_MASK	0xff
+#define VLV_FW_PIPEB_SC_WM1_MASK	0xff00
 #define VLV_FW_PIPEB_SD_MASK	0xff0000
+#define VLV_FW_PIPEB_SD_WM1_MASK	0xff000000
 #define VLV_FW_PIPEB_SD_SHIFT	16
 #define VLV_DSPHOWM_PIPEB_MASK	0x111000
 #define VLV_DSPHOWM_PIPEB_PB_SHIFT	12
@@ -4299,9 +4305,12 @@ enum punit_power_well {
 #define VLV_DSPHOWM_PIPEB_SD_SHIFT	20
 
 #define VLV_FW_PIPEC_PC_MASK	0xff0000
+#define VLV_FW_PIPEC_PC_WM1_MASK	0xff000000
 #define VLV_FW_PIPEC_PC_SHIFT	16
 #define VLV_FW_PIPEC_SE_MASK	0xff
+#define VLV_FW_PIPEC_SE_WM1_MASK	0xff00
 #define VLV_FW_PIPEC_SF_MASK	0xff0000
+#define VLV_FW_PIPEC_SF_WM1_MASK	0xff000000
 #define VLV_FW_PIPEC_SF_SHIFT	16
 #define VLV_DSPHOWM_PIPEC_MASK	0xe00000
 #define VLV_DSPHOWM_PIPEC_PC_SHIFT	21
@@ -4312,6 +4321,7 @@ enum punit_power_well {
 #define VLV_FW_PIPEB_WM1_MASK2	0xff000000
 
 #define VLV_FW_SR_MASK		0xff800000
+#define VLV_FW_SR_WM1_MASK	0xff
 #define VLV_DSPHOWM_SR_MASK	0x3000000
 /* drain latency register values*/
 #define VLV_DDL(pipe)	_PIPE(pipe, VLV_DDL1, VLV_DDL2)
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index a5dfc83..af252f5 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1684,6 +1684,7 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 
 	if (sr) {
 		I915_WRITE_BITS(DSPFW1, (sr << 23), VLV_FW_SR_MASK);
+		I915_WRITE_BITS(DSPFW6, 0, VLV_FW_SR_WM1_MASK);
 		I915_WRITE(DSPHOWM, (I915_READ(DSPHOWM) &
 						~(VLV_DSPHOWM_SR_MASK)) |
 					((sr >> 9) << 24));
@@ -1699,6 +1700,9 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 		I915_WRITE_BITS(DSPFW2, sa, VLV_FW_PIPEA_SA_MASK);
 		I915_WRITE_BITS(DSPFW2, (sb << VLV_FW_PIPEA_SB_SHIFT),
 							VLV_FW_PIPEA_SB_MASK);
+		I915_WRITE_BITS(DSPFW5, 0, VLV_FW_PIPEA_PA_WM1_MASK);
+		I915_WRITE_BITS(DSPFW4, 0, VLV_FW_PIPEA_SA_WM1_MASK);
+		I915_WRITE_BITS(DSPFW4, 0, VLV_FW_PIPEA_SB_WM1_MASK);
 		I915_WRITE(DSPHOWM, (I915_READ(DSPHOWM) &
 						~(VLV_DSPHOWM_PIPEA_MASK)) |
 						(((pa >> 8) | ((sa >> 8) <<
@@ -1714,9 +1718,12 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 		/* update wm */
 		I915_WRITE_BITS(DSPFW1, (pa << VLV_FW_PIPEB_PB_SHIFT),
 							VLV_FW_PIPEB_PB_MASK);
+		I915_WRITE_BITS(DSPFW5, 0, VLV_FW_PIPEB_PB_WM1_MASK);
 		I915_WRITE_BITS(DSPFW7, sa, VLV_FW_PIPEB_SC_MASK);
+		I915_WRITE_BITS(DSPFW7, 0, VLV_FW_PIPEB_SC_WM1_MASK);
 		I915_WRITE_BITS(DSPFW7, (sb << VLV_FW_PIPEB_SD_SHIFT),
 							VLV_FW_PIPEB_SD_MASK);
+		I915_WRITE_BITS(DSPFW7, 0, VLV_FW_PIPEB_SD_WM1_MASK);
 		I915_WRITE(DSPHOWM, (I915_READ(DSPHOWM) &
 					~(VLV_DSPHOWM_PIPEB_MASK)) |
 				((((pa >> 8) << VLV_DSPHOWM_PIPEB_PB_SHIFT) |
@@ -1731,9 +1738,12 @@ void vlv_update_dsparb(struct intel_crtc *intel_crtc)
 		/* update wm */
 		I915_WRITE_BITS(DSPFW9, (pa << VLV_FW_PIPEC_PC_SHIFT),
 							VLV_FW_PIPEC_PC_MASK);
+		I915_WRITE_BITS(DSPFW9, 0, VLV_FW_PIPEC_PC_WM1_MASK);
 		I915_WRITE_BITS(DSPFW8, sa, VLV_FW_PIPEC_SE_MASK);
+		I915_WRITE_BITS(DSPFW8, 0, VLV_FW_PIPEC_SE_WM1_MASK);
 		I915_WRITE_BITS(DSPFW8, (sb << VLV_FW_PIPEC_SF_SHIFT),
 							VLV_FW_PIPEC_SF_MASK);
+		I915_WRITE_BITS(DSPFW8, 0, VLV_FW_PIPEC_SE_WM1_MASK);
 		I915_WRITE(DSPHOWM, (I915_READ(DSPHOWM) &
 						~(VLV_DSPHOWM_PIPEC_MASK)) |
 				((((pa >> 8) << VLV_DSPHOWM_PIPEC_PC_SHIFT) |
-- 
1.7.9.5

